### DISCUSSION TITLE
Improvements for the server list window

### DISCUSSION THREAD

**Discussion by LC-Zorg** [OpenTTD/OpenTTD#9751] [Features]
Out of the many interface elements, I haven't touched this window so far. I didn't see particular needs for changes here, but when considering a solution to the problem with the "Add server" button, a few ideas came to mind. I don't know if they are good, but maybe something will be useful. :)

![New server list window](https://user-images.githubusercontent.com/67662046/145911419-c787766e-0dae-48ec-bbe8-65df0d406068.png)

Description of changes:
1. The window separately displays three types of servers:
- Public (internet)
- Local (LAN)
- Private (invite only)
2. There are no "Search internet" and "Search LAN" buttons - their function is replaced by the above server type selection list and the "Refresh" button.
3. The "Add server" button function is completely removed - the server can be found in the same field at the top based on the invitation code or IP address. There is no dedicated field for entering the invitation code. I think it is unnecessary - when in the server type selection part it says "Private (invite only)", the function of the "Search" part seems clear enough. The tooltips would contain the relevant information in case they do.
4. According to its name, the new "Add server" button is used to add your own server to the list. After pressing, a selection window appears: the player can choose whether he wants to continue the previous online game (if one has been saved) or whether he wants to start a new game.
![Add new server (bb)](https://user-images.githubusercontent.com/67662046/176052172-f2637c01-6241-4b36-bd49-9386f945ba26.png)
5. There is one change in the creating a new game window - the player can see if and what password has been set - this is currently quite problematic.
6. An additional change in the main window is adding markings for favorite servers. This is the same solution I presented for the NewGRF settings window. To change the designation, just press LMB on the star, and it will change the color and status of the server. Favorite servers would always be on top.

**Comment by James103** [comment]
What's the difference between the gold star, the silver star, and no star in the first column of the listing?

**Reply by LC-Zorg** [reply]
These stars indicate the order of the marked servers. The gold ones are at the top, the silver ones are below, and the unmarked servers are  further on. You click LMB on the star and it changes color: none> silver> gold> none> ...

**Comment by LC-Zorg** [comment]
A few other ideas inspired by a suggestion some time ago [posted on tt-forums](https://www.tt-forums.net/viewtopic.php?p=1253418#p1253418) (adding many (too many imho) filters, making it easier to spot servers based on other versions of the game, especially JGRPP) ...

1. Adding a server filter bar according to the nature of the gameplay - I don't know, maybe I'm wrong, the points of view may be different, but I think that among the filters, if you limit them to one, the most useful one could be the one that defines the type of gameplay on the server. In the adding server window, there would be an appropriate setting that the server administrator could but would not have to use.
2. Marking servers based on the JGRPP version with colored text placed in front of the server name - Administrators often do not include this phrase in the name, which makes searching for these servers very difficult. Perhaps this is more of a solution that JGR could use, but in the official version, that wouldn't be a bad thing either.
![New server list window v2 3b](https://user-images.githubusercontent.com/67662046/176050693-b82007b1-37cf-482e-a97f-f60df9f02208.png)

3. Showing the game version in place of the colored dots - In this solution the meaning of the text color would remain unchanged. It would be more of a statistical curiosity in most situations, but in case someone wants to play a specific version of the game, it would be very helpful.
![New server list window v2 4c](https://user-images.githubusercontent.com/67662046/176050715-1cf58def-ed02-42a0-aa52-6f79828026de.png)

Regarding point 1. In the City Mania client, the player has "tabs" listed for popular groups of servers such as CityMania, reddit, N-ice, BTPro. I find it good, although it may be debatable that it favors certain servers this way.
![CityMania bar](https://user-images.githubusercontent.com/67662046/176051363-14c9a4e0-9d1a-426c-84c3-9c6121b295f5.png)
Perhaps allowing the player to create their own tabs could be a useful solution.
